bigquery/storage/managedwriter/retry.go

```diff
@@ -21,17 +21,26 @@ import (
 	"time"
 
 	"github.com/googleapis/gax-go/v2"
+	"github.com/googleapis/gax-go/v2/apierror"
+	storagepb "google.golang.org/genproto/googleapis/cloud/bigquery/storage/v1"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+var (
+	defaultAppendRetries = 3
+	knownReconnectErrors = []error{
+		io.EOF,
+		status.Error(codes.Unavailable, "the connection is draining"), // errStreamDrain in gRPC transport
```

Comment:
https://screenshot.googleplex.com/3PHj8Dh8iq2VoNk

Reply:
These are the fast-reconnect predicates, where we force a close+reopen on the AppendRows bidi.  In the case of drain we close quickly rather than letting more appends get enqueued on a draining connection.

Reply:
Refactored this.  We'll look for io.EOF, but the Unavailable will get handled under the more-generic code-based retry.